notification/index.googleFunction.js

```diff
@@ -24,8 +24,8 @@ exports.notificationTrigger = async (request, response) => {
       { notification: getNotificationForTracking(notificationItems), err },
       'Unexpected exception occurred.'
     )
-    if (err.isRecoverable) {
-      return response.status(400).send(err.message)
+    if (!err.isRecoverable) {
+      return response.status(500).send(err.message)
```

Comment:
Did not we say that we return 500 on recoverable errors only? On non recoverable it has to be `accepted` so that we do not block other notifications.

Reply:
With this implementation, I changed the meaning of recoverable, if it’s recoverable it will be accepted, if not it will be rejected. Was it better before ? 

In this case: 
- All validation errors will be `err.isRecoverable = true` means will return "accepted" to Adyen service.
- Unexpected errors like bugs will be `err.isRecoverable = undefined` means will return 500 to Adyen service.
- All `commercetools` related API errors like 400 will be `err.isRecoverable=true` means will return "accepted" to Adyen service.
- Errors like NetworkError (statusCode=0) or API errors like statusCode=5xx will be `err.isRecoverable ` means will return 500 to Adyen service.

Reply:
To be honest I find it confusing.

> All validation errors will be err.isRecoverable = true means will return "accepted" to Adyen service.

In this case from integration perspective it is not recoverable but because we do not want to block we return OK/accepted. It think previously it was easier to understand the intention of recoverable vs. non recoverable.

> Unexpected errors like bugs will be err.isRecoverable = undefined means will return 500 to Adyen service

Unexpected errors should lead to accepted and not 500, right?

> All commercetools related API errors like 400 will be err.isRecoverable=true means will return "accepted" to Adyen service.

Again, it is confusing as from integration perspective we have an obviously not recoverable 400 error.

Reply:
Could you please check again: https://github.com/commercetools/commercetools-adyen-integration/pull/629/commits/46a544d55af0abd27920873f58617b3f06a61217